src/io/windows/sync.rs

```diff
@@ -0,0 +1,238 @@
+        debug_assert!(offset + buf.len() as u64 <= self.size());
+        let file_inner = self.inner.clone();
+        Self::background_sync_call(move || file_inner.write_all_at(&buf, offset)).await
+    }
+
+    pub(crate) async fn read_all(&self) -> Result<BytesMut> {
+        self.read_exact_at_allocate(self.size().try_into()?, 0).await
+    }
+
+    pub(crate) async fn read_exact_at_allocate(
+        &self,
+        size: usize,
+        offset: u64,
+    ) -> Result<BytesMut> {
+        let buf = BytesMut::zeroed(size);
+        self.read_exact_at(buf, offset).await
+    }
+
+    pub(crate) async fn read_exact_at(&self, mut buf: BytesMut, offset: u64) -> Result<BytesMut> {
+        let file_inner = self.inner.clone();
+        Self::background_sync_call(move || file_inner.read_exact_at(&mut buf, offset).map(|_| buf)).await
+            .map_err(|err| err.into())
+    }
+
+    pub(crate) async fn fsyncdata(&self) -> IOResult<()> {
+        let file_inner = self.inner.clone();
+        Self::background_sync_call(move || file_inner.std_file.sync_all()).await
+    }
+
+    pub(crate) fn created_at(&self) -> IOResult<SystemTime> {
+        let metadata = self.inner.std_file.metadata()?;
+        Ok(metadata.created().unwrap_or(SystemTime::now()))
+    }
+
+    async fn background_sync_call<F, R>(f: F) -> R
+    where
+        F: FnOnce() -> R + Send + 'static,
+        R: Send + 'static,
+    {
+        tokio::task::spawn_blocking(move || f())
+            .await
+            .expect("spawned blocking task failed")
+    }
+
+
+
+    fn from_file(
+        path: impl AsRef<Path>,
+        setup: impl Fn(&mut OpenOptions) -> &mut OpenOptions,
+    ) -> IOResult<Self> {
+        const FILE_SHARE_READ: u32 = 1;
+        let file = 
+            match setup(&mut OpenOptions::new().share_mode(FILE_SHARE_READ)).open(path.as_ref()) {
+                Ok(file) => file,
+                Err(err) if err.raw_os_error().unwrap_or_default() == 32 => {
+                    error!("File {:?} is locked", path.as_ref());
+                    panic!("File {:?} is locked", path.as_ref());
+                },
+                Err(err) => {
+                    return Err(err);
+                }
+            };
+
+        let size = file.metadata()?.len();
+        let size = AtomicU64::new(size);
+
+        Ok(Self {
+            inner: Arc::new(FileInner {
+                std_file: file,
+                size
+            })
+        })
+    }
+}
+
+
+impl FileInner {
+    fn read_exact_at(&self, mut buf: &mut [u8], mut offset: u64) -> IOResult<()> {
+        while !buf.is_empty() {
+            match self.std_file.seek_read(buf, offset) {
+                Ok(0) => break,
+                Ok(n) => {
+                    let tmp = buf;
+                    buf = &mut tmp[n..];
+                    offset += n as u64;
+                }
+                Err(ref e) if e.kind() == std::io::ErrorKind::Interrupted => {}
```

Comment:
`std::io::{Error, ErrorKind}` should be imported

Reply:
This is common type names, so I've imported them with `std::io::{Error as IOError, ErrorKind as IOErrorKind}`